gym/lib/gym/options.rb

```diff
@@ -104,6 +122,9 @@ def self.plain_options
                                        if value
                                          UI.important "Don't use this option any more, as it's deprecated by Apple"
                                        end
+                                       if Gym::Xcode.legacy_api_deprecated?
+                                         Gym::Options.legacy_api_note
```

Comment:
At fastlane very often we use `method_name!` with a `!` to clarify that this is either modifying the object, or raising an exception if something goes wrong, could you rename the method?